### ISSUE TITLE
Explicit termination method 'end' not called on Android

### ISSUE THREAD

**Issue by everwanna** [srikanth-lingala/zip4j#127]
Similar to #3 and #17 

It looks like `inflater.end()` not called in `InflaterInputStream`.

```java
2020-01-11 21:48:13.555 23005-23020/tech.bluespace.android.id_guard_test D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
        at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1877)
        at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:286)
        at java.util.zip.Inflater.finalize(Inflater.java:407)
        at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:289)
        at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:276)
        at java.lang.Daemons$Daemon.run(Daemons.java:137)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.Throwable: Explicit termination method 'end' not called
        at dalvik.system.CloseGuard.open(CloseGuard.java:237)
        at java.util.zip.Inflater.<init>(Inflater.java:122)
        at net.lingala.zip4j.io.inputstream.InflaterInputStream.<init>(InflaterInputStream.java:20)
        at net.lingala.zip4j.io.inputstream.ZipInputStream.initializeDecompressorForThisEntry(ZipInputStream.java:220)
        at net.lingala.zip4j.io.inputstream.ZipInputStream.initializeEntryInputStream(ZipInputStream.java:201)
        at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:106)
        at net.lingala.zip4j.tasks.AbstractExtractFileTask.verifyNextEntry(AbstractExtractFileTask.java:81)
        at net.lingala.zip4j.tasks.AbstractExtractFileTask.extractFile(AbstractExtractFileTask.java:47)
        at net.lingala.zip4j.tasks.ExtractAllFilesTask.executeTask(ExtractAllFilesTask.java:35)
        at net.lingala.zip4j.tasks.ExtractAllFilesTask.executeTask(ExtractAllFilesTask.java:13)
        at net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:42)
        at net.lingala.zip4j.tasks.AsyncZipTask.execute(AsyncZipTask.java:36)
        at net.lingala.zip4j.ZipFile.extractAll(ZipFile.java:437)
```

**Comment by srikanth-lingala** [comment]
Can you share your code and the circumstance in which the application was terminated? Was the application abruptly terminated or did the application run was complete, in other words, was the application force terminated or did zip4j complete its task before the application stopped?

**Comment by srikanth-lingala** [comment]
I think I found the issue. I made a fix for it. Fix is not in master but in the [inflater-end][1] branch. I just did a SNAPSHOT release with version `2.3.1-SNAPSHOT` with the fix. Can you please use this version and see if this fixes the issue? You have to follow the instruction mentioned [here][2] to get this snapshot version in your maven.

[1]: https://github.com/srikanth-lingala/zip4j/tree/inflater-end
[2]: https://stackoverflow.com/a/7717234/2736153

**Comment by everwanna** [comment]
Good, the error is gone with version 2.3.1-SNAPSHOT. Thanks man!

**Comment by srikanth-lingala** [comment]
Thanks for the confirmation. I will include the fix in the next release.

**Comment by srikanth-lingala** [comment]
Fixed in v2.3.1 released today.
